src/driver/sqlserver/SqlServerQueryRunner.ts

```diff
@@ -1505,126 +1510,165 @@ export class SqlServerQueryRunner extends BaseQueryRunner implements QueryRunner
+                      ISNULL(Objectproperty(Object_id(CONCAT(TABLE_CATALOG, '.', TABLE_SCHEMA, '.', TABLE_NAME)), 'IsMSShipped'), 0) = 0
+                `;
+            }).join(" UNION ALL ");
+            dbTables.push(...await this.query(tablesSql));
+        } else {
+            const dbNames = tableNames
+                .filter(tablePath => tablePath.split(".").length === 3)
+                .map(tablePath => tablePath.split(".")[0]);
+            if (this.driver.database && !dbNames.find(dbName => dbName === this.driver.database))
+                dbNames.push(this.driver.database);
+
+            const tablesCondition = tableNames.map(tableName => {
+                let [database, schema, name] = tableName.split(".");
+                // if name is empty, it means that tableName have only schema name and table name or only table name
+                if (!name) {
+                    // if schema is empty, it means tableName have only name of a table. Otherwise it means that we have "schemaName"."tableName" string.
+                    if (!schema) {
+                        name = database;
+                        schema = this.driver.options.schema || currentSchema;
 
-                } else {
-                    name = schema;
-                    schema = database;
+                    } else {
+                        name = schema;
+                        schema = database;
+                    }
+                } else if (schema === "") {
+                    schema = this.driver.options.schema || currentSchema;
                 }
-            } else if (schema === "") {
-                schema = this.driver.options.schema || currentSchema;
-            }
 
-            return [schema, name];
-        };
 
-        tableNames.filter(tablePath => tablePath.indexOf(".") !== -1)
-            .forEach(tablePath => {
-                if (tablePath.split(".").length === 3) {
-                    if (tablePath.split(".")[1] !== "")
-                        schemaNames.push(tablePath.split(".")[1]);
-                } else {
-                    schemaNames.push(tablePath.split(".")[0]);
-                }
-            });
-        schemaNames.push(this.driver.options.schema || currentSchema);
+                return `("TABLE_SCHEMA" = '${schema}' AND "TABLE_NAME" = '${name}')`;
+            }).join(" OR ");
 
-        const dbNames = tableNames
-            .filter(tablePath => tablePath.split(".").length === 3)
-            .map(tablePath => tablePath.split(".")[0]);
-        if (this.driver.database && !dbNames.find(dbName => dbName === this.driver.database))
-            dbNames.push(this.driver.database);
+            const tablesSql = dbNames.map(dbName => {
+                return `
+                    SELECT
+                        "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME"
+                    FROM "${dbName}"."INFORMATION_SCHEMA"."TABLES"
+                    WHERE "TABLE_TYPE" = 'BASE TABLE' AND ${tablesCondition}`;
```

Comment:
Later we use `sys.tables`, probably should here too for consistency

Reply:
Oh, nope. `INFORMATION_SCHEMA` is the ANSI wrapper.  Let's stick with that instead.